packages/core/src/internal/proc.js

```diff
@@ -40,21 +40,19 @@ function getIteratorMetaInfo(iterator, fn) {
   return getMetaInfo(fn)
 }
 
-// TODO: check if this hacky toString stuff is needed
-// also check again whats the difference between CHANNEL_END and CHANNEL_END_TYPE
+// toString is used to have ability to log actions in devtools.
 // maybe this could become MAYBE_END
 // I guess this gets exported so takeMaybe result can be checked
-export const CHANNEL_END = {
-  toString() {
-    return CHANNEL_END_SYMBOL
-  },
-}
+
 export const TASK_CANCEL = {
   toString() {
     return TASK_CANCEL_SYMBOL
   },
 }
 
+const shouldTerminate = (res) => res && (isEnd(res) || res[TERMINATE] || res === TASK_CANCEL)
```

Comment:
I get a feeling that something is off with our `prettier`/`lint-staged` config, single arrow param should be reformatted without parens 🤔

this needs to be investigated, but not as part of this PR